lib/p2p/P2P.js

```diff
@@ -43,19 +56,23 @@ class P2P extends EventEmitter {
    */
   connect(host, port) {
     return new Promise((resolve, reject) => {
-      dns.lookup(host, (dnsErr, ipAddr) => {
+      dns.lookup(host, (dnsErr, address) => {
         if (dnsErr) {
           resolve(`could not resolve ${host}`);
+          return;
         }
-        if (this.peers[`${ipAddr}:${port}`]) {
-          resolve(`peer ${host}:${port} is already connected`);
+        const peerId = P2P.getPeerId(host, address, port);
+        if (this.peers[peerId]) {
+          resolve(`peer ${peerId} is already connected`);
+          return;
         }
+
         const peer = net.connect({
-          host,
+          address,
           port,
         }, () => {
-          this.addPeer(peer);
-          const msg = `connected to peer ${peer.remoteAddress}:${peer.remotePort}`;
+          this.addPeer(peer, peerId);
+          const msg = `connected to peer ${peerId}`;
```

Comment:
Here we still have access to the `host` variable. Maybe we do something like:

```javascript
const msg = host ? `connected to peer (${host})${peerId}` : `connected to peer ${peerId}`
```

to display the host name if one is provided, otherwise just the ip address and port.